src/bin/tikv-server.rs

```diff
@@ -206,10 +205,12 @@ fn run_raft_server(pd_client: RpcClient, cfg: &TiKvConfig, security_mgr: Arc<Sec
 
     let engines = Engines::new(Arc::new(kv_engine), Arc::new(raft_engine));
 
-    let storage_read_pool =
-        ReadPool::new("store-read", &cfg.readpool.storage.build_config(), || {
-            storage::ReadPoolContext::new(pd_sender.clone())
-        });
+    let storage_read_pool = storage::ReadPoolImpl::build_read_pool(
```

Comment:
Personally I prefer `storage::ReadPool`. `Impl` looks like a private thing.

Reply:
`ReadPool` had been used, maybe use `ReadPoolProducer`. Is it ok?

Reply:
Or `ReadPoolContext`? It just can build a `ReadPool` and handle some metrics. It's not a `ReadPool` indeed.

Reply:
It just "derive"s the common ReadPool to create a *specialized* ReadPool that attached some name, some lifetime hook functions (like on_tick). That's why it was called `ReadPoolImpl`. `Producer` or `Builder` might not be a very good name because it will be confusing for functions like `Producer:: tls_collect_executor_metrics`.

Reply:
Agree with `Producer` and `Builder` are not good enough. How about remove the struct?